### PR TITLE
Rebuild for LLVM 16

### PR DESCRIPTION
**Pull request by h-vetinari**

### CODE REVIEW

**Comment by h-vetinari** [comment]
Hm, something not working together with clang + configure:
```
configure:4553: $BUILD_PREFIX/bin/clang -V >&5
clang-16: error: argument to '-V' is missing (expected 1 value)
clang-16: error: no input files
configure:4564: $? = 1
configure:4553: $BUILD_PREFIX/bin/clang -qversion >&5
clang-16: error: unknown argument '-qversion'; did you mean '--version'?
clang-16: error: no input files
configure:4564: $? = 1
configure:4553: $BUILD_PREFIX/bin/clang -version >&5
clang-16: error: unknown argument '-version'; did you mean '--version'?
clang-16: error: no input files
```

**Comment by h-vetinari** [comment]
Well no, configuring has lots of expected failures while testing what's available.

This seems to be the actual error:
```
2023-02-28T06:16:02.0818148Z /home/conda/feedstock_root/build_artifacts/cctools-and-ld64_1677564625459/work/cctools/ld64/src/3rd/qsort_r.c:247:1: warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
2023-02-28T06:16:02.0819862Z swapfunc(a, b, n, swaptype)
2023-02-28T06:16:02.0825966Z ^
2023-02-28T06:16:02.0839295Z /home/conda/feedstock_root/build_artifacts/cctools-and-ld64_1677564625459/work/cctools/ld64/src/3rd/qsort_r.c:247:1: warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
2023-02-28T06:16:02.1346217Z /home/conda/feedstock_root/build_artifacts/cctools-and-ld64_1677564625459/work/cctools/ld64/src/3rd/helper.c:45:5: error: call to undeclared function '__assert'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
2023-02-28T06:16:02.1348377Z     __assert(msg, file, line);
2023-02-28T06:16:02.1348834Z     ^
2023-02-28T06:16:02.1359487Z 1 error generated.
2023-02-28T06:16:02.1412905Z make[4]: *** [Makefile:475: libhelper_la-helper.lo] Error 1
```

**Comment by h-vetinari** [comment]
Which is weird, because that file has the right [include](https://github.com/tpoechtrager/cctools-port/blob/973.0.1-ld64-609/cctools/ld64/src/3rd/helper.c#L19).

**Comment by h-vetinari** [comment]
And glic does [declare](https://github.com/bminor/glibc/blob/glibc-2.12/assert/assert.h#L80-L83) that function since forever...

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -29,3 +22,20 @@ cxx_compiler_version:   # [osx]
...
   - 14.0
   - 15.0
+  - 16.0
+
+# no channel_targets necessary because LLVM rc's not installable without llvm_rc anyway;
+# zip it to avoid using rc-builds for already-released-in-conda-forge LLVM versions
+channel_sources:
+  - conda-forge
+  - conda-forge
+  - conda-forge/label/llvm_rc,conda-forge
```
Comment (h-vetinari) [inline recipe/conda_build_config.yaml]: My thinking is: if we _don't_ add the channel_targets, we wouldn't have to rebuild cctools after the rc-phase; the build of cctools against 16.0.0.rc3 would continue to work against 16.0.x GA.

**Review by h-vetinari — COMMENTED** [review]
@isuruf, I managed to figure this one out. Not sure if you agree with my reasoning to publish this without a label; if not, I'm happy to add appropriate channel_targets as well!

**Comment by h-vetinari** [comment]
Could you PTAL here and let me know what you think @isuruf?

File: recipe/build.sh [hunk 2]
```diff
@@ -55,7 +54,5 @@ pushd cctools_build_final
     --disable-static \
     --with-libtapi=${PREFIX} \
     --enable-shared || (cat config.log && cat config.status && false)
-  cat config.log
-  cat config.status
-  make -j${CPU_COUNT} ${VERBOSE_AT} -k
+  make -j${CPU_COUNT} ${VERBOSE_AT}
```
Comment (isuruf) [inline recipe/build.sh]: Please revert these changes

File: recipe/conda_build_config.yaml (hunk 1 above)
Comment (isuruf) [reply recipe/conda_build_config.yaml]: Let's not do that. conda/mamba solvers prefers the latest version and if the latest version is not installable, then all bets are off and users might get very old versions to satisfy some constraint.

File: recipe/build.sh (hunk 2 above)
Comment (h-vetinari) [reply recipe/build.sh]: For context: https://github.com/conda-forge/cctools-and-ld64-feedstock/commit/28196b162f033296d154d8c666c39c6427707b14 was explicitly for debugging purposes; it spams the logs quite extensively (I would have thought it's good enough to show them in the error case).

And `-k` makes it much harder to chase down the actual error, because the build keeps going until it can't (forcing to scroll through tons of unrelated logs before finding actual errors).

I would have considered both these things an improvement, but if you want...

Comment (isuruf) [reply recipe/build.sh]: Yes, they were for debugging purposes, but keeping them there is not much of a burden and helps debug the next error easily.

File: recipe/conda_build_config.yaml (hunk 1 above)
Comment (h-vetinari) [reply recipe/conda_build_config.yaml]: OK, I considered this, but thought it would probably work. But fine to add the channel targets.

File: recipe/build.sh (hunk 2 above)
Comment (h-vetinari) [reply recipe/build.sh]: I can see the marginally positive utility of the `cat`s, but for `-k`, it's IMO really more of a burden than a help. In any case, I did as you asked.

**Comment by h-vetinari** [comment]
Thanks for the quick feedback! This should hopefully look better / be ready now.

File: recipe/build.sh (hunk 2 above)
Comment (isuruf) [reply recipe/build.sh]: Yeah, I don't see any usefulness of `-k`. Sorry about selecting all three lines
